pandas/core/dtypes/cast.py

```diff
@@ -1602,7 +1602,8 @@ def construct_1d_object_array_from_listlike(values: Sized) -> np.ndarray:
     # numpy will try to interpret nested lists as further dimensions, hence
     # making a 1D array that contains list-likes is a bit tricky:
     result = np.empty(len(values), dtype="object")
-    result[:] = values
+    for i, obj in enumerate(values):
```

Comment:
Any advantage of using `np.fromiter(values, dtype="object", count=len(values))`?

Reply:
Ah, nice, wasn't aware of that. From a quick test that seems to be even a bit faster